Clarify that Setter sets values only. (#976)
open-telemetry/opentelemetry-specification: specification/context/api-propagators.md

Optional arguments:

> - A `Setter` invoked to set each propagation key/value pair. This is an additional

Reviewer: ```suggestion
- A `Setter` invoked to set a propagation key/value pair. This is an additional
```

Why "each"? Is there some kind of iteration involved?

Author: That's for formats using more than one header (like the old B3 format with multiple entries).

Reviewer: I understand that the setter can be invoked for more than one header, or even a dynamic set of headers. But I think just "each" is prone to misunderstanding.

Would the following clarification be better?

```suggestion
- A `Setter` that, each time it is invoked, sets the given propagation key/value pair. This is an additional
```

> - A `Setter` to set a propagation key/value pair every time it is invoked. This is an additional
>   argument that languages are free to define to help inject data into the carrier.

Reviewer 2: Alternatively:
```suggestion
- A `Setter` to set a propagation key/value pair. It might be invoked multiple times in order to set multiple pairs.
  This is an additional argument that languages are free to define to help inject data into the carrier.
```

> Optional arguments:

Reviewer: I don't think setter is an optional argument in languages that choose to use it.

Reviewer 2: We can fix this in #986.

> - A `Setter` to set a propagation key/value pair. It might be invoked multiple times in order to set multiple pairs.

Reviewer: ```suggestion
- A `Setter` to set a propagation key/value pair. Propagators MAY invoke it multiple times in order to set multiple pairs.
```

>   This is an additional argument that languages are free to define to help inject data into the carrier.

Reviewer: If we have this sentence *and* the "Optional arguments:" above, I fear this may lead to misunderstandings.

Author: @arminru ^ 😉

Reviewer 2: --> #986

Reviewer: @arminru will you take this?

Reviewer 2: Yep, will do.

> #### Setter argument

Reviewer: Actually I am confused by this spec: Does this heading describe the arguments to the setter? Should this be heading be replaced by the sentence: "Inject should have the following parameters:"?

- A `Context`.
- The carrier that holds propagation fields.

> - The `Setter` invoked to set each propagation key/value pair.

Reviewer: ```suggestion
- The `Setter` invoked to set a propagation key/value pair.
```

> - The `Setter` to set a propagation key/value pair every time it is invoked.

Reviewer 2: ```suggestion
- The `Setter` to set a propagation key/value pair. It might be invoked multiple times in order to set multiple pairs.
```

> - The `Setter` to set a propagation key/value pair. It might be invoked multiple times in order to set multiple pairs.

Reviewer: ```suggestion
- The `Setter` to set a propagation key/value pair. Propagators MAY invoke it multiple times in order to set multiple pairs.
```
